zeppelin-zengine/src/test/java/org/apache/zeppelin/notebook/NotebookTest.java

```diff
@@ -150,13 +150,14 @@ public void testSchedule() throws InterruptedException, IOException{
 		note.setConfig(config);
 		notebook.refreshCron(note.id());
 		Thread.sleep(1*1000);
-		dateFinished = p.getDateFinished();
-		assertNotNull(dateFinished);
-
+		
 		// remove cron scheduler.
 		config.put("cron", null);
 		note.setConfig(config);
 		notebook.refreshCron(note.id());
+		Thread.sleep(1000);
+		dateFinished = p.getDateFinished();
+		assertNotNull(dateFinished);
 		Thread.sleep(1*1000);
 		assertEquals(dateFinished, p.getDateFinished());
```

Comment:
The logic is that if the notebook has stopped running, the p.getDateFinished in line 162 will have the same result as the one in line 159. If the paragraph is running, the date returned in 162 will have advanced by one second over the call in 159 and the test will fail.

The only difference over the original code, is that I moved the tests further behind the config.put. The actual dates checked don't really matter, therefore the delayed check for dateFinished with the pause does not impact the test. If the paragraph didn't stop, the test will still fail the assertion in 162.